docs/website/docs/general-usage/schema-contracts.md

```diff
@@ -90,34 +91,34 @@ Note that this works in two directions. If you use a model with such setting exp
 | discard_value | ignore         |
 | discard_row   | forbid         |
 
-`discard_row` requires additional handling when ValidationError is raised.
+`discard_row` requires additional handling when a ValidationError is raised.
 
 :::tip
 Model validation is added as a [transform step](resource.md#filter-transform-and-pivot-data) to the resource. This step will convert the incoming data items into instances of validating models. You could easily convert them back to dictionaries by using `add_map(lambda item: item.dict())` on a resource.
 :::
 
 :::note
-Pydantic models work on the **extracted** data **before names are normalized or nested tables are created**. Make sure to name model fields as in your input data and handle nested data with the nested models.
+Pydantic models work on the **extracted** data **before names are normalized or nested tables are created**. Make sure to name model fields as in your input data and handle nested data with nested models.
 
-As a consequence, `discard_row` will drop the whole data item - even if nested model was affected.
+As a consequence, `discard_row` will drop the whole data item - even if a nested model was affected.
 :::
 
-### Set contracts on Arrow Tables and Pandas
+### Set contracts on Arrow tables and Pandas
+
 All contract settings apply to [arrow tables and panda frames](../dlt-ecosystem/verified-sources/arrow-pandas.md) as well.
-1. **tables** mode the same - no matter what is the data item type
-2. **columns** will allow new columns, raise an exception or modify tables/frames still in extract step to avoid re-writing parquet files.
-3. **data_type** changes to data types in tables/frames are not allowed and will result in data type schema clash. We could allow for more modes (evolving data types in Arrow tables sounds weird but ping us on Slack if you need it.)
+1. **tables** mode is the same - no matter what the data item type is.
+2. **columns** will allow new columns, raise an exception, or modify tables/frames still in the extract step to avoid rewriting parquet files.
```

Comment:
```suggestion
2. **columns** will allow new columns, raise an exception, or modify tables/frames still in the extract step to avoid rewriting Parquet files.
```